### ISSUE TITLE
Release 4.6.0

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1331]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] Address all coding and style violations detected during a *Release* build (send a regular pull request which must be merged before continuing)
- [x] ensure completeness of draft [release notes](https://github.com/FakeItEasy/FakeItEasy/releases), including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
- [x] run `tools\prepare_release.cmd x.y.z` to:
    - change "vNext" names to appropriate x.y.z version for the next GitHub Release, milestone, and release checklist (this issue)
    - create a new draft GitHub Release, milestone, and release checklist
    - create release/x.y.z branch on upstream, branching from develop
    - submit a pull request to merge the release/x.y.z branch to master
- [x] merge the pull request, _which may be done by the PR creator, unlike a regular pull request_
- [x] tag master with the new version number and push to upstream. This will cause the build server to
    - push the nupkgs to [nuget.org](https://www.nuget.org/profiles/FakeItEasy) and
    - attach all nupkgs from the tagged build to the GitHub Release.
- [x] wait for NuGet package validation
- [x] de-list pre-release NuGet packages if present
- [x] publish the GitHub release
- [x] tweet, mentioning contributors, and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [x] close this issue
- [ ] close the [milestone](https://github.com/FakeItEasy/FakeItEasy/milestones?state=open) for this release

**Comment by blairconrad** [comment]
I'd be inclined to release as soon as we have something that can provide relief for #1345. I've a PR in flight that would resolve #1348, which doesn't _directly_ address #1345, but does catch a broader class of Dummy creation failures.

**Comment by thomaslevesque** [comment]
👍

**Comment by blairconrad** [comment]
Tweet:

```
Get FakeItEasy 4.6.0!
Fixed:
  Exceptions caught while resolving constructor arguments break Fake/Dummy creation (symptom: unable to create Dummy string under .NET Core 2.1-RC1).
Thanks, Ryan Wischkaemper.
  
https://github.com/FakeItEasy/FakeItEasy/releases/4.6.0
```

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.6.0](https://github.com/FakeItEasy/FakeItEasy/releases/4.6.0).
